File: Systems-for-Design-and-Implementation/Labs/Lab11/ProiectCSharp/TransportPersistence/transport.persistence/ISeatRepository.cs
using System.Collections.Generic;
using Transport.Model.transport.model;

namespace TransportPersistence.transport.persistence
{
    public interface ISeatRepository : IRepository<long, Seat>
    {
        IEnumerable<Seat> FindSeatsByReservationId(long reservationId);

        List<Seat> FindSeatsByTripId(long tripId);
    }
}

File: Systems-for-Design-and-Implementation/Labs/Lab12/ProiectJava/TransportPersistence/src/main/java/transport/persistence/ISeatRepository.java
package transport.persistence;

import transport.model.Seat;

import java.util.List;

public interface ISeatRepository extends Repository<Long, Seat> {

    Iterable<Seat> findSeatsByReservationId(long reservationId);

    List<Seat> findSeatsByTripId(Long tripId);
}
